bindings/experimental/distrdf/python/DistRDF/Ranges.py

```diff
@@ -13,16 +14,28 @@ class EmptySourceRange(object):
 
     Attributes:
 
+    id (int): Sequential counter to identify this range. It is used to assign
+        a filename to a partial Snapshot in case it was requested. The id
+        counter is reset to zero whenever a new RDataFrame is created in the
+        Python session.
+
     start (int): Starting entry of this range.
 
     end (int): Ending entry of this range.
     """
 
+    _ids = count()
+
     def __init__(self, start, end):
         """set attributes"""
+        self.id = next(self._ids)
         self.start = start
         self.end = end
 
+    @classmethod
+    def reset_index(cls):
+        """Reset index of EmptySourceRange objects in this run."""
+        cls._ids = count()
```

Comment:
the fact we need this class method means that the class itself does not have enough information to take care of the range count by itself ("someone else" needs to remember to tell it to reset the count).

would it be a simpler design to simply pass the id to `__init__`?

Reply:
Probably, but it would take a refactor of the `get_clustered_ranges` function. Maybe an `enumerate` is enough, not sure

Reply:
if it's just adding a counter or an `enumerate` in `get_clustered_ranges`, that seems like a simpler design?